Userland/Libraries/LibC/CMakeLists.txt

```diff
@@ -76,6 +76,26 @@ set(LIBC_SOURCES
     wstdio.cpp
 )
 
+file(GLOB_RECURSE LIBC_HEADERS RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}" CONFIGURE_DEPENDS "*.h")
+list(APPEND LIBC_HEADERS "../LibELF/ELFABI.h" "../LibRegex/RegexDefs.h")
+
+add_custom_target(install_libc_headers)
+
+# Copy LibC's headers into the sysroot to satisfy libc++'s include priority requirements.
+foreach(RELATIVE_HEADER_PATH IN LISTS LIBC_HEADERS)
+    get_filename_component(filename ${RELATIVE_HEADER_PATH} NAME)
+    get_filename_component(directory ${RELATIVE_HEADER_PATH} DIRECTORY)
+    string(REPLACE "../" "" subdirectory "${directory}")
+    file(MAKE_DIRECTORY "${CMAKE_STAGING_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}/${subdirectory}")
+    add_custom_command(
+        TARGET install_libc_headers
```

Comment:
if I'm reading this right, this will create a distinct custom command target for each header in LibC? that sounds like... a lot of commands. Is there a way we can do this in one command? I feel like this will be kind of slow. 

At the very least, we should make sure that the custom prebuild command has a dependency on the input file location, and that the custom target `install_libc_headers` has a dependency on the destination file location.

Reply:
i.e., have you tested that updating any LibC header will cause this action to re-run?

Reply:
Right... install(FILES) can work on directories with a FILES_MATCHING pattern, not sure about file(COPY) at configure time or any of the things we can do in the build stage rather than the install or configure stage.